client/executor/src/native_executor.rs

```diff
@@ -269,12 +290,138 @@ impl<D: NativeExecutionDispatch> RuntimeInfo for NativeExecutor<D> {
 			runtime_code,
 			ext,
 			false,
-			|_instance, version, _ext|
+			|_module, _instance, version, _ext|
 				Ok(version.cloned().ok_or_else(|| Error::ApiError("Unknown version".into()))),
 		)
 	}
 }
 
+/// Helper inner struct to implement `RuntimeSpawn` extension.
+pub struct RuntimeInstanceSpawn {
+	module: Arc<dyn WasmModule>,
+	forks: parking_lot::Mutex<HashMap<u64, mpsc::Receiver<Vec<u8>>>>,
```

Comment:
Nit: could we shift this terminology to something that uses `spawn`? First, it could suggest that it is a fork while it is actually not (at least comparing with the `fork` syscall), second it seems that the code is already settled on the term `spawn`.